packages/ember-data/lib/serializers/json-api-serializer.js

```diff
@@ -0,0 +1,294 @@
+/**
+  @module ember-data
+*/
+
+import Serializer from "ember-data/system/serializer";
+
+var copy = Ember.copy;
+var get = Ember.get;
+var forEach = Ember.ArrayPolyfills.forEach;
+var map = Ember.ArrayPolyfills.map;
+
+/**
+  @class JSONAPISerializer
+  @namespace DS
+  @extends DS.Serializer
+*/
+export default Serializer.extend({
+
+  keyForAttribute: function(key) {
+    return Ember.String.dasherize(key);
+  },
+
+  keyForRelationship: function(key, kind) {
+    return Ember.String.dasherize(key);
+  },
+
+  normalizeTypeKey: function(typeKey) {
+    return Ember.String.camelize(typeKey);
+  },
```

Comment:
Could this be changed to `Ember.String. dasherize(typeKey)`? If that doesn't work today I expect it will be something we want to do in the future as we move to dasherized type keys internality.